csharp_dotnetcore/Using-Cards/Using_Cards/CardsBot.cs

```diff
@@ -0,0 +1,339 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Dialogs.Choices;
+using Microsoft.Bot.Schema;
+
+namespace Using_Cards
+{
+    public class CardsBot : IBot
+    {
+        private readonly CardsBotAccessors _accessors;
+        public DialogSet Dialogs;
+
+        public CardsBot(CardsBotAccessors accessors)
+        {
+            _accessors = accessors;
+
+            Dialogs = new DialogSet(accessors.ConversationDialogState);
+            Dialogs.Add(new WaterfallDialog("cardSelector", new WaterfallStep[] { ChoiceCardStep, ShowCardStep }));
+            Dialogs.Add(new ChoicePrompt("cardPrompt"));
+        }
+
+        public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = new CancellationToken())
+        {
+            switch (turnContext.Activity.Type)
+            {
+                case ActivityTypes.Message:
+                    try
+                    {
+                        var dc = await Dialogs.CreateContextAsync(turnContext, cancellationToken);
+                        await dc.ContinueAsync(cancellationToken);
+
+                        if (!dc.Context.Responded)
+                        {
+                            await dc.BeginAsync("cardSelector", cancellationToken: cancellationToken);
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine(e);
```

Comment:
This doesn't make sense. This isn't a console bot. Remove the try/catch altogether.